test/shared.js

```diff
@@ -144,4 +144,54 @@ X-Poedit-SourceCharset: UTF-8`;
       expect(folded.length).to.equal(3);
     });
   });
+
+  describe('parseNPluralFromHeadersSafely', () => {
+    it('should return parsed value', () => {
+      const headers = { 'Plural-Forms': 'nplurals=10; plural=n' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(10);
+    });
+
+    it('should return parsed value (missing plural declaration)', () => {
+      const headers = { 'Plural-Forms': 'nplurals=10' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(10);
+    });
+
+    it('should return fallback value ("Plural-Forms" header is absent)', () => {
+      const nplurals = parseNPluralFromHeadersSafely();
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (nplurals is not declared)', () => {
+      const headers = { 'Plural-Forms': '; plural=n' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (nplurals is set to zero)', () => {
+      const headers = { 'Plural-Forms': 'nplurals=0' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (nplurals is set to negative value)', () => {
+      const headers = { 'Plural-Forms': 'nplurals=-99' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (failed to parse nplurals value)', () => {
```

Comment:
Should it return fallback value in this case or rather throw an error? I would opt for the latter